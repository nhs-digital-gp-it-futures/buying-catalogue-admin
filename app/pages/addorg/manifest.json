{
  "title": "Find an organisation",
  "description": "Enter the correct Organisation Data Service (ODS) code to find the organisation you're looking for.",
  "isMultiQuestion": true,
  "questions": [
    {
      "id": "odsCode",
      "mainAdvice": "Enter ODS code"
    }
  ],
  "continueButtonText": "Continue"
}
