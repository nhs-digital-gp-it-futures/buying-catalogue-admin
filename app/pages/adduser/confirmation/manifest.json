{
  "backLinkText": "Back to dashboard",
  "backLinkHref": "/organisations/{{organisationId}}",
  "title": "{{userAdded}} account added",
  "description": "The user has been sent an email so they can set their password."
}
