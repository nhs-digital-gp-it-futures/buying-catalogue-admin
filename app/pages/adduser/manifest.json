{
  "title": "Add new user",
  "description": "Provide the following information to make a new user account.",
  "orgNameSubheading": "Organisation name",
  "isMultiQuestion": true,
  "questions": [
    {
      "id": "firstName",
      "mainAdvice": "First name"
    },
    {
      "id": "lastName",
      "mainAdvice": "Last name"
    },
    {
      "id": "phoneNumber",
      "mainAdvice": "Telephone number"
    },
    {
      "id": "emailAddress",
      "mainAdvice": "Email address"
    }
  ],
  "addUserButtonText": "Add user"
}
