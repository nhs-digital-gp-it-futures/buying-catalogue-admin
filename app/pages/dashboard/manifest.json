{
  "title": "Manage organisations and users",
  "description": "Edit existing account information or add new user to an organisation.",
  "columnInfo": [
    {
      "data": "Organisation name"
    },
    {
      "data": "ODS code"
    }
  ],
  "columnClass": "nhsuk-grid-column-one-half"
}
