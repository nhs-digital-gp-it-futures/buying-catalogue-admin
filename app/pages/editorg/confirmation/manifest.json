{
  "backLinkText": "Back to dashboard",
  "title": "{{organisationName}} updated",
  "description": "This organisation's details have been updated.",
  "dataTestId": "edit-organisation-confirmation"
}
