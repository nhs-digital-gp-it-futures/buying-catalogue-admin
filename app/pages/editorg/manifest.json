{
  "description": "Change the Organisation End User Agreement status for this organisation.",
  "odsCodeHeading": "ODS code",
  "organisationNameHeading": "Organisation name",
  "primaryRoleIdHeading": "Primary Role ID",
  "addressHeading": "Business address",
  "catalogueAgreementCheckboxText": "Organisation End User Agreement has been signed",
  "saveButtonText": "Save"
}
