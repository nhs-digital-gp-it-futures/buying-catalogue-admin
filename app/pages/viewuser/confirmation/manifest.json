{
  "enable": {
    "backLinkText": "Back to user",
    "title": "{{userName}} account re-enabled",
    "description": "The user account has been successfully rectivated.",
    "dataTestId": "enable-user-confirmation"
  },
  "disable": {
    "backLinkText": "Back to user",
    "title": "{{userName}} account disabled",
    "description": "This user account is no longer active.",
    "dataTestId": "disable-user-confirmation"
  }
}
