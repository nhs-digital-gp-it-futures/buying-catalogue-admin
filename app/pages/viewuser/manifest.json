{
  "title": "details",
  "orgHeading": "Organisation name",
  "nameHeading": "Name",
  "phoneNumberHeading": "Contact details",
  "emailAddressHeading": "Email address"
}
