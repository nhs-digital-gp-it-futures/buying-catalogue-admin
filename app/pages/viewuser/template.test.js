import { componentTester, snapshotTest } from '../../test-utils/componentTester';
import manifest from './manifest.json';

const setup = {
  template: {
    path: 'pages/viewuser/template.njk',
  },
};

const mockData = {
  organisationId: 'org1',
  organisationName: 'Greater Manchester CCG',
  userName: 'John Smith',
  phoneNumber: '07777777777',
  emailAddress: 'John@Smith.com',
  accountDisabled: false,
  changeAccountStatusButtonText: 'Disable account',
  changeAccountStatusFormAction: '/organisations/org1/user2/disable',
};

const mockContext = {
  ...manifest,
  ...mockData,
  backLinkHref: '/organisations/org1',
  csrfToken: 'mockCsrfToken',
};

describe('viewuser page', () => {
  it('the page should render', componentTester(setup, (harness) => {
    harness.request(mockContext, ($) => {
      const snapshot = snapshotTest($, '[data-test-id="main-content"]');
      expect(snapshot).toMatchSnapshot();
    });
  }));

  it('should render a backLink to the organisation page', componentTester(setup, (harness) => {
    harness.request(mockContext, ($) => {
      const homepageBackLink = $('[data-test-id="go-back-link"]');
      expect(homepageBackLink.length).toEqual(1);
      expect(homepageBackLink.text().trim()).toEqual('Go back');
      expect($(homepageBackLink).find('a').attr('href')).toEqual(mockContext.backLinkHref);
    });
  }));

  it('should render a change account status button', componentTester(setup, (harness) => {
    harness.request(mockContext, ($) => {
      const changeAccountStatusButton = $('[data-test-id="change-account-status-button"] button');
      expect(changeAccountStatusButton.length).toEqual(1);
      expect(changeAccountStatusButton.text().trim())
        .toEqual(mockContext.changeAccountStatusButtonText);
    });
  }));

  it('should create form element with correct action', componentTester(setup, (harness) => {
    harness.request(mockContext, ($) => {
      const formElement = $('form');
      expect(formElement.length).toEqual(1);
      expect(formElement.attr('action')).toEqual(mockContext.changeAccountStatusFormAction);
    });
  }));

  it('should render hidden input with csrf token', componentTester(setup, (harness) => {
    harness.request(mockContext, ($) => {
      const formElement = $('input[name=_csrf]');
      expect(formElement.length).toEqual(1);
      expect(formElement.attr('type')).toEqual('hidden');
      expect(formElement.attr('value')).toEqual(mockContext.csrfToken);
    });
  }));
});
