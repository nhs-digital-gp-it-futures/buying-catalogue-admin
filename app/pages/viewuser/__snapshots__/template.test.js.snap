// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viewuser page the page should render 1`] = `
"<main class=\\"nhsuk-main-wrapper\\" id=\\"maincontent\\" data-test-id=\\"main-content\\">
        
  <div class=\\"nhsuk-width-container\\" data-test-id=\\"view-user-page\\">
    <div data-test-id=\\"go-back-link\\">
      <div class=\\"nhsuk-back-link\\">
  <a class=\\"nhsuk-back-link__link\\" href=\\"/organisations/org1\\">
  <svg class=\\"nhsuk-icon nhsuk-icon__chevron-left\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\">
    <path d=\\"M8.5 12c0-.3.1-.5.3-.7l5-5c.4-.4 1-.4 1.4 0s.4 1 0 1.4L10.9 12l4.3 4.3c.4.4.4 1 0 1.4s-1 .4-1.4 0l-5-5c-.2-.2-.3-.4-.3-.7z\\"></path>
  </svg>
  Go back</a>
</div>

    </div>
  
    <div class=\\"nhsuk-grid-row\\">
      <div class=\\"nhsuk-grid-column-two-thirds\\">
        <h1 class=\\"nhsuk-u-margin-top-7\\" data-test-id=\\"view-user-page-title\\">John Smith details</h1>
      </div>
    </div>

    <dl class=\\"nhsuk-summary-list nhsuk-u-padding-margin-5\\">
      <div class=\\"nhsuk-summary-list__row\\">
        <dt class=\\"nhsuk-summary-list__key\\">
          Organisation name
        </dt>
        <dd class=\\"nhsuk-summary-list__value\\">
          Greater Manchester CCG
        </dd>
      </div>
      <div class=\\"nhsuk-summary-list__row\\">
        <dt class=\\"nhsuk-summary-list__key\\">
          Name
        </dt>
        <dd class=\\"nhsuk-summary-list__value\\">
          John Smith
        </dd>
      </div>
      <div class=\\"nhsuk-summary-list__row\\">
        <dt class=\\"nhsuk-summary-list__key\\">
          Contact details
        </dt>
        <dd class=\\"nhsuk-summary-list__value\\">
          07777777777
        </dd>
      </div>
      <div class=\\"nhsuk-summary-list__row\\">
        <dt class=\\"nhsuk-summary-list__key\\">
          Email address
        </dt>
        <dd class=\\"nhsuk-summary-list__value\\">
          John@Smith.com
        </dd>
      </div>
    </dl>

    <div class=\\"nhsuk-grid-row\\">
      <form method=\\"post\\" action=\\"/organisations/org1/user2/disable\\">
        <input type=\\"hidden\\" name=\\"_csrf\\" value=\\"mockCsrfToken\\">
        <div data-test-id=\\"change-account-status-button\\" class=\\"nhsuk-grid-column-one-quarter\\">
          


  
    
  




<button class=\\"nhsuk-button nhsuk-u-margin-bottom-9\\" type=\\"submit\\">
  Disable account
</button>



        </div>
      </form>
    </div>
  <div>

      </div></div></main>"
`;
